File: tests/fixtures/good.java
/*
 * Copyright 2026 René El Fatuaki
 * Licensed under the MIT License
 * https://opensource.org/license/mit
 */
package example;

class Good {

  /** Returns the customer ID, or null when the session is anonymous. */
  String customerId() { return null; }

  /**
   * Retries idempotent requests with exponential backoff.
   * @see <a href="https://docs.spring.io/spring-framework/reference/integration/rest-clients.html">Spring REST clients</a>
   */
  void send() {
    // Resolve the token first so a retry never races the credential store.
    // https://docs.spring.io/spring-security/reference/servlet/authentication/index.html
    int token = 1;
    // TODO: remove the retry cap #88
    System.out.println(token);
  }
}

File: tests/fixtures/good.kt
/*
 * Copyright 2026 René El Fatuaki
 * Licensed under the MIT License
 * https://opensource.org/license/mit
 */
package example

/** Returns the user for `id`, or null once the cache has expired. */
fun cachedUser(id: Int): Int? = id

/**
 * Collects the flow only while the lifecycle is at least STARTED.
 * [repeatOnLifecycle](https://developer.android.com/topic/libraries/architecture/coroutines#restart-lifecycle)
 */
fun observeUser() {
    // Resolve the token first so a retry never races the keystore.
    // https://developer.android.com/training/articles/keystore
    val token = 1
    // TODO: drop the fallback APP-123
    println(token)
}

val raw = """
  a — b; // still a string
"""
